src/planner/binder/query_node/bind_select_node.cpp

```diff
@@ -380,10 +384,31 @@ unique_ptr<BoundQueryNode> Binder::BindSelectNode(SelectNode &statement, unique_
 			auto bound_expr = group_binder.Bind(group_expressions[i], &group_type);
 			D_ASSERT(bound_expr->return_type.id() != LogicalTypeId::INVALID);
 
+			// find out whether the expression contains a subquery, it can't be copied if so
+			auto &bound_expr_ref = *bound_expr;
+			bool contains_subquery = bound_expr_ref.GetExpressionClass() == ExpressionClass::BOUND_SUBQUERY;
+			ExpressionIterator::EnumerateChildren(bound_expr_ref, [&](const Expression &child) {
+				contains_subquery = contains_subquery || child.GetExpressionClass() == ExpressionClass::BOUND_SUBQUERY;
+			});
+
 			// push a potential collation, if necessary
-			bound_expr = ExpressionBinder::PushCollation(context, std::move(bound_expr),
-			                                             StringType::GetCollation(group_type), true);
-			result->groups.group_expressions.push_back(std::move(bound_expr));
+			auto collated_expr = ExpressionBinder::PushCollation(context, std::move(bound_expr),
+			                                                     StringType::GetCollation(group_type), true);
+			if (!contains_subquery && !collated_expr->Equals(bound_expr_ref)) {
+				// if there is a collation on a group x, we should group by the collated expr,
+				// but also push a first(x) aggregate in case x is selected (uncollated)
+				info.collated_groups[i] = result->aggregates.size();
+
+				auto first_fun = FirstFun::GetFunction(LogicalType::VARCHAR);
+				vector<unique_ptr<Expression>> first_children;
+				// FIXME: would be better to just refer to this expression, but for now we copy
+				first_children.push_back(bound_expr_ref.Copy());
```

Comment:
I have been trying something for a while now, but I'm starting to think that I don't understand what you mean.

This is what I tried:
```c++
			// push a potential collation, if necessary
			string collation = StringType::GetCollation(group_type);
			auto collated_expr = ExpressionBinder::PushCollation(context, std::move(bound_expr), collation, true);
			if (!collation.empty()) {
				// if a collation is added on a group x, we should group by the collated expr,
				// but also push a first(x) aggregate in case x is selected (uncollated)
				vector<unique_ptr<ParsedExpression>> first_children;
				if (group_binder.unbound_expression->GetExpressionClass() == ExpressionClass::COLLATE) {
					auto &collate_expr = group_binder.unbound_expression->Cast<CollateExpression>();
					first_children.push_back(collate_expr.child->Copy());
				} else {
					first_children.push_back(group_binder.unbound_expression->Copy());
				}
				auto first = make_uniq_base<ParsedExpression, FunctionExpression>("first", std::move(first_children));
				result->aggregates.push_back(group_binder.Bind(first));
				info.collated_groups[i] = result->aggregates.size() - 1;
			}
```
But the binder doesn't seem to like this.

Reply:
Hm yes, that is what I meant. Let's just keep the original implementation if that's not working.